src/rocdecode-host/rocdecode_host_api.cpp

```diff
@@ -0,0 +1,200 @@
+//! Destroy the decoder object
+/*****************************************************************************************************/
+rocDecStatus ROCDECAPI 
+rocDecDestroyDecoderHost(rocDecDecoderHandle decoder_handle) {
+    if (decoder_handle == nullptr) {
+        return ROCDEC_INVALID_PARAMETER;
+    }
+    auto handle = static_cast<DecHandleHost *>(decoder_handle);
+    delete handle;
+    return ROCDEC_SUCCESS;
+}
+
+/**********************************************************************************************************************/
+//! \fn rocDecStatus ROCDECAPI rocDecGetDecoderCapsHost(rocDecDecoderHandle decoder_handle, RocdecDecodeCaps *pdc)
+//! Queries decode capabilities of host based decoder based on codec type, chroma_format and BitDepthMinus8 parameters.
+//! 1. Application fills IN parameters codec_type, chroma_format and BitDepthMinus8 of RocdecDecodeCaps structure
+//! 2. On calling rocdecGetDecoderCaps, driver fills OUT parameters if the IN parameters are supported
+//!    If IN parameters passed to the driver are not supported by AMD-VCN-HW, then all OUT params are set to 0.
+//!    Currently this is not supported in avcodec based decoder
+/**********************************************************************************************************************/
+rocDecStatus ROCDECAPI
+rocDecGetDecoderCapsHost(RocdecDecodeCaps *pdc) {
+    if (pdc == nullptr) {
+        return ROCDEC_INVALID_PARAMETER;
+    }
+    //todo:
+    return ROCDEC_NOT_IMPLEMENTED;
+}
+
+/*****************************************************************************************************/
+//! \fn rocDecStatus ROCDECAPI rocDecDecodeFrameHost(rocDecDecoderHandle decoder_handle, RocdecPicParams *pic_params)
+//! Decodes a single picture
+//! Submits the frame for HW decoding 
+/*****************************************************************************************************/
+rocDecStatus ROCDECAPI 
+rocDecDecodeFrameHost(rocDecDecoderHandle decoder_handle, _RocdecPicParamsHost *pic_params) {
+    if (decoder_handle == nullptr || pic_params == nullptr) {
+        return ROCDEC_INVALID_PARAMETER;
+    }
+    auto handle = static_cast<DecHandleHost *>(decoder_handle);
+    rocDecStatus ret;
+    try {
+        ret = handle->roc_decoder_host_->DecodeFrame(pic_params);
+    }
+    catch(const std::exception& e) {
+        handle->CaptureError(e.what());
+        ERR(e.what())
+        return ROCDEC_RUNTIME_ERROR;
+    }
+    return ret;
+}
+
+/************************************************************************************************************/
+//! \fn rocDecStatus ROCDECAPI rocDecGetDecodeStatusHost(rocDecDecoderHandle decoder_handle, int pic_idx, RocdecDecodeStatus* decode_status);
+//! Get the decode status for frame corresponding to pic_idx
+//! API is currently supported for HEVC codec.
+//! API returns CUDA_ERROR_NOT_SUPPORTED error code for unsupported GPU or codec.
```

Comment:
Please change to ROCDEC_ERROR_NOT_SUPPORTED and remove `CUDA` in the comments section